src/umap.cpp

```diff
@@ -58,6 +57,8 @@ class _umap {
         void* segment_address;
         size_t segment_length;
         int backingfile_fd;
+        int num_bk_files;
+        umap_backing_file* bk_files;
```

Comment:
While the API can use a structure pointer since it is a "C" interface, I would like to have this be a vector<umap_backing_file> bk_files;  num_bk_files can be removed and bk_files.size() can be used instead.